pkg/batch/subscriber/processor/domain_event_informer.go

```diff
@@ -44,161 +40,31 @@ var (
 	ErrUnknownSourceType = errors.New("batch-server: domain-event-informer unknown source type")
 )
 
-type options struct {
-	metrics metrics.Registerer
-	logger  *zap.Logger
-}
-
-var defaultOptions = options{
-	logger: zap.NewNop(),
-}
-
-type Option func(*options)
-
-func WithMetrics(r metrics.Registerer) Option {
-	return func(opts *options) {
-		opts.metrics = r
-	}
-}
-
-func WithLogger(logger *zap.Logger) Option {
-	return func(opts *options) {
-		opts.logger = logger
-	}
-}
-
-type domainEventInformer struct {
-	environmentClient            environmentclient.Client
-	sender                       sender.Sender
-	maxMPS                       int
-	runningDurationPerBatch      time.Duration
-	project                      string
-	domainSubscription           string
-	domainTopic                  string
-	pullerNumGoroutines          int
-	pullerMaxOutstandingMessages int
-	pullerMaxOutstandingBytes    int
-	opts                         *options
-	logger                       *zap.Logger
+type DomainEventInformer struct {
```

Comment:
How about creating an interface to return `Process(msg *puller.Message)` and make the struct private?
Wdyt?

Reply:
Great idea! Let me implement this change.